src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -1740,13 +1744,18 @@ def create_ddos_plan(cmd, resource_group_name, ddos_plan_name, location=None, ta
     from azure.cli.core.commands import LongRunningOperation
 
     ddos_client = network_client_factory(cmd.cli_ctx).ddos_protection_plans
+    ddos_protection_plan = cmd.get_models('DdosProtectionPlan')()
+    if location:
+        ddos_protection_plan.location = location
+    if tags:
+        ddos_protection_plan.tags = tags
     if not vnets:
         # if no VNETs can do a simple PUT
-        return ddos_client.create_or_update(resource_group_name, ddos_plan_name, location=location, tags=tags)
+        return ddos_client.begin_create_or_update(resource_group_name, ddos_plan_name, parameters=ddos_protection_plan)
 
     # if VNETs specified, have to create the protection plan and then add the VNETs
     plan_id = LongRunningOperation(cmd.cli_ctx)(
-        ddos_client.create_or_update(resource_group_name, ddos_plan_name, location=location, tags=tags)).id
+        ddos_client.begin_create_or_update(resource_group_name, ddos_plan_name, parameters=ddos_protection_plan)).id
```

Comment:
In track2, SDK parameters are not flattened, so we need pass object parameter instead of flattened parameters to SDK.